### PR TITLE
Remove winapi as a dependency

### PR DESCRIPTION
**Pull request by rylev**

### CODE REVIEW

File: examples/basic/server/src/british_short_hair_cat.rs [hunk 1]
```diff
@@ -14,21 +13,21 @@ pub struct BritishShortHairCat {
 impl IDomesticAnimal for BritishShortHairCat {
     unsafe fn train(&self) -> HRESULT {
         println!("Training...");
-        NOERROR
+        0
```
Comment (snf) [inline examples/basic/server/src/british_short_hair_cat.rs]: Keep the naming?

File: src/inproc.rs [hunk 2]
```diff
@@ -60,11 +47,14 @@ pub fn unregister_keys(registry_keys_to_remove: Vec<RegistryKeyInfo>) -> HRESULT
     hr
 }
 
-fn create_class_key(key_info: &RegistryKeyInfo) -> Result<HKEY, LSTATUS> {
-    let hk_result: HKEY = std::ptr::null_mut::<c_void>() as HKEY;
-    let lp_class = std::ptr::null_mut::<CHAR>();
-    let lp_security_attributes = std::ptr::null_mut::<SECURITY_ATTRIBUTES>();
-    let lpdw_disposition = std::ptr::null_mut::<DWORD>();
+const HKEY_CLASSES_ROOT: *mut c_void = 0x8000_0000 as *mut c_void;
```
Comment (snf) [inline src/inproc.rs]: Pretty sure this should be u32

Comment (snf) [reply src/inproc.rs]: `typedef DWORD HKEY` in the header

File: src/inproc.rs [hunk 3]
```diff
@@ -108,7 +99,7 @@ fn set_class_key(key_handle: HKEY, key_info: &RegistryKeyInfo) -> Result<HKEY, L
     Ok(key_handle)
 }
 
-fn add_class_key(key_info: &RegistryKeyInfo) -> LSTATUS {
+fn add_class_key(key_info: &RegistryKeyInfo) -> i32 {
```
Comment (snf) [inline src/inproc.rs]: Keep the name?

File: src/runtime.rs [hunk 4]
```diff
@@ -80,21 +71,21 @@ impl ApartmentThreadedRuntime {
         clsid: &IID,
         outer: &mut U,
     ) -> Result<InterfacePtr<T>, HRESULT> {
-        unsafe { self.create_raw_instance::<T>(clsid, outer as *mut U as LPUNKNOWN) }
+        unsafe { self.create_raw_instance::<T>(clsid, outer as *mut U as *mut *const c_void) }
```
Comment (snf) [inline src/runtime.rs]: Extra pointer modifier or IUnknown would be (*const void)?

File: src/sys.rs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    pub data1: u32,
+    pub data2: u16,
+    pub data3: u16,
+    pub data4: [u8; 8],
+}
+
+#[link(name = "ole32")]
+extern "system" {
+    pub fn RegCreateKeyExA(
+        hKey: *mut c_void,
```
Comment (snf) [inline src/sys.rs]: hkey is DWORD/unsigned long

File: src/sys.rs [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        dwOptions: u32,
+        samDesired: u32,
+        lpSecurityAttributes: *mut c_void,
+        phkResult: *mut *mut c_void,
+        lpdwDisposition: *mut u32,
+    ) -> i32;
+    pub fn GetModuleFileNameA(hModule: *mut c_void, lpFilename: *mut i8, nSize: u32) -> u32;
+    pub fn RegCloseKey(hKey: *mut c_void) -> i32;
+    pub fn RegSetValueExA(
+        hKey: *mut c_void,
```
Comment (snf) [inline src/sys.rs]: same

File: src/sys.rs [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        lpValueName: *const i8,
+        Reserved: u32,
+        dwType: u32,
+        lpData: *const u8,
+        cbData: u32,
+    ) -> i32;
+    pub fn RegDeleteKeyA(hKey: *mut c_void, lpSubKey: *const i8) -> i32;
```
Comment (snf) [inline src/sys.rs]: same

File: examples/aggregation/server/src/local_file_manager.rs [hunk 8]
```diff
@@ -13,7 +12,7 @@ pub struct LocalFileManager {
 impl ILocalFileManager for LocalFileManager {
     unsafe fn delete_local(&self) -> HRESULT {
         println!("Deleting Locally...");
-        NOERROR
+        0
```
Comment (snf) [inline examples/aggregation/server/src/local_file_manager.rs]: keep name

File: examples/aggregation/server/src/windows_file_manager.rs [hunk 9]
```diff
@@ -18,7 +17,7 @@ pub struct WindowsFileManager {
 impl IFileManager for WindowsFileManager {
     unsafe fn delete_all(&self) -> HRESULT {
         println!("Deleting all by delegating to Local and Remote File Managers...");
-        NOERROR
+        0
```
Comment (snf) [inline examples/aggregation/server/src/windows_file_manager.rs]: keep name

File: examples/basic/server/src/british_short_hair_cat.rs [hunk 10]
```diff
@@ -14,21 +13,21 @@ pub struct BritishShortHairCat {
...
     }
 }
 
 impl ICat for BritishShortHairCat {
     unsafe fn ignore_humans(&self) -> HRESULT {
         println!("Ignoring Humans...");
-        NOERROR
+        0
```
Comment (snf) [inline examples/basic/server/src/british_short_hair_cat.rs]: same

File: examples/basic/server/src/british_short_hair_cat.rs [hunk 11]
```diff
@@ -14,21 +13,21 @@ pub struct BritishShortHairCat {
...
     }
 }
 
 impl IAnimal for BritishShortHairCat {
     unsafe fn eat(&self) -> HRESULT {
         println!("Eating...");
-        NOERROR
+        0
```
Comment (snf) [inline examples/basic/server/src/british_short_hair_cat.rs]: same

File: src/inproc.rs (hunk 2 above)
Comment (rylev) [reply src/inproc.rs]: The examples no longer work when I make this change.

Comment (snf) [reply src/inproc.rs]: Compile or runtime? Weird

**Pull request merged**